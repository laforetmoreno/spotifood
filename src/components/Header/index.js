import React from 'react';
import { array, func, object, string, oneOfType, number, instanceOf } from 'prop-types';
import { useMediaQuery } from 'react-responsive';

import DatePickerWrapper from 'components/DatePickerWrapper';
import Select from 'components/Select';
import Input from 'components/Input';
import Collapse from 'components/Collapse';

import { limitExemple, mobileSize } from '../../constants';

import styles from './index.module.scss';

const Header = ({
  filters,
  onLocaleChange,
  onCountryChange,
  onLimitChange,
  onDateChange,
  onInputChange,
  limit,
  country,
  locale,
  startDate,
  playlistName,
}) => {
  const isMobile = useMediaQuery({ query: mobileSize });

  const renderContent = () => (
    <ul className={styles.wrapper}>
      <Select
        className={styles.select}
        placeholder={locale?.value}
        options={filters[0]?.values}
        onChange={onLocaleChange}
        label="Locale"
      />
      <Select
        className={styles.select}
        placeholder={country?.name}
        options={filters[1]?.values}
        onChange={onCountryChange}
        label="Country"
      />
      <Select
        className={styles.select}
        placeholder={limit}
        options={limitExemple}
        onChange={onLimitChange}
        label="Quantity"
      />
      <Input
        placeholder="Relax"
        onChange={onInputChange}
        value={playlistName}
        label="Playlist name"
      />
      <DatePickerWrapper onChange={onDateChange} value={startDate} />
    </ul>
  );

  const renderPage = () => {
    if (isMobile) {
      return <Collapse>{renderContent()}</Collapse>;
    }
    return renderContent();
  };

  return renderPage();
};

Header.propTypes = {
  filters: array.isRequired,
  onLocaleChange: func,
  onCountryChange: func,
  onLimitChange: func,
  onDateChange: func,
  onInputChange: func,
  limit: oneOfType([string, number]),
  country: object,
  locale: object,
  startDate: instanceOf(Date),
  playlistName: string,
};

Header.defaultProps = {
  onLocaleChange: () => '',
  onCountryChange: () => '',
  onLimitChange: () => null,
  onDateChange: () => new Date(),
  onInputChange: () => '',
  limit: 0,
  country: {},
  locale: {},
  startDate: new Date(),
  playlistName: '',
};

export default Header;
