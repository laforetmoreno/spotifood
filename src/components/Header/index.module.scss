@import '../../styles';

.wrapper {
  padding: rem(20px);
  display: flex;
  align-items: inherit;
  justify-content: space-evenly;
  max-width: 960px;
  margin: 0 auto;

  @media (max-width: 767px) {
    @include flex-column();
    padding: rem(10px) 0;
  }
}
