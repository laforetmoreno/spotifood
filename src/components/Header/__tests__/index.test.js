import React from 'react';
import renderer from 'react-test-renderer';
import { render, waitForElement, fireEvent } from '@testing-library/react';
import Header from '../index';

describe('<Header />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('get a input element', async () => {
    const { getByPlaceholderText } = render(<Header />);
    const fieldNode = await waitForElement(() => getByPlaceholderText('----'));
    fireEvent.change(fieldNode);
  });
});
