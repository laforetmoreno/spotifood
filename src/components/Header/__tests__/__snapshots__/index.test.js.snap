// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<ul
  className="wrapper"
>
  <div
    className="wrapper"
  >
    <span
      className="label"
    >
      Locale
    </span>
    <select
      className="select select false"
      data-testid="select"
      defaultValue="default"
      disabled={false}
      name=""
      onChange={[Function]}
    >
      <option
        value="default"
      >
        
      </option>
    </select>
  </div>
  <div
    className="wrapper"
  >
    <span
      className="label"
    >
      Country
    </span>
    <select
      className="select select false"
      data-testid="select"
      defaultValue="default"
      disabled={false}
      name=""
      onChange={[Function]}
    >
      <option
        value="default"
      >
        
      </option>
    </select>
  </div>
  <div
    className="wrapper"
  >
    <span
      className="label"
    >
      Quantity
    </span>
    <select
      className="select select false"
      data-testid="select"
      defaultValue="default"
      disabled={false}
      name=""
      onChange={[Function]}
    >
      <option
        value="default"
      >
        0
      </option>
      <option
        value={5}
      >
        5
      </option>
      <option
        value={10}
      >
        10
      </option>
      <option
        value={20}
      >
        20
      </option>
      <option
        value={50}
      >
        50
      </option>
    </select>
  </div>
  <div
    className="wrapper"
  >
    <span
      className="label"
    >
      Playlist name
    </span>
    <input
      className="input"
      data-testid="input"
      name=""
      onChange={[Function]}
      placeholder="Relax"
      value=""
    />
  </div>
  <div
    className="wrapper"
  >
    <div
      className="label"
    >
      Date
    </div>
    <div
      className="react-date-picker react-date-picker--closed react-date-picker--enabled picker"
      onFocus={[Function]}
    >
      <div
        className="react-date-picker__wrapper"
      >
        <div
          className="react-date-picker__inputGroup"
          onClick={[Function]}
        >
          <input
            max="5760-09-12"
            min="1821-04-19"
            name="date"
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "left": "-9999px",
                "position": "absolute",
                "top": "-9999px",
                "visibility": "hidden",
              }
            }
            type="date"
            value="2020-09-03"
          />
          
          <input
            autoComplete="off"
            className="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
            max={12}
            min={1}
            name="month"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="--"
            required={null}
            type="number"
            value={9}
          />
          <span
            className="react-date-picker__inputGroup__divider"
          >
            /
          </span>
          <input
            autoComplete="off"
            autoFocus={false}
            className="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
            max={30}
            min={1}
            name="day"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="--"
            required={null}
            type="number"
            value={3}
          />
          <span
            className="react-date-picker__inputGroup__divider"
          >
            /
          </span>
          <input
            autoComplete="off"
            autoFocus={false}
            className="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
            max={275760}
            min={1}
            name="year"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="----"
            required={null}
            step={1}
            type="number"
            value={2020}
          />
          
        </div>
        <button
          className="react-date-picker__clear-button react-date-picker__button"
          onClick={[Function]}
          onFocus={[Function]}
          type="button"
        >
          <svg
            className="react-date-picker__clear-button__icon react-date-picker__button__icon"
            height={19}
            stroke="black"
            strokeWidth={2}
            viewBox="0 0 19 19"
            width={19}
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="4"
              x2="15"
              y1="4"
              y2="15"
            />
            <line
              x1="15"
              x2="4"
              y1="4"
              y2="15"
            />
          </svg>
        </button>
        <button
          className="react-date-picker__calendar-button react-date-picker__button"
          onClick={[Function]}
          onFocus={[Function]}
          type="button"
        >
          <svg
            className="react-date-picker__calendar-button__icon react-date-picker__button__icon"
            height={19}
            stroke="black"
            strokeWidth={2}
            viewBox="0 0 19 19"
            width={19}
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="15"
              width="15"
              x="2"
              y="2"
            />
            <line
              x1="6"
              x2="6"
              y1="0"
              y2="4"
            />
            <line
              x1="13"
              x2="13"
              y1="0"
              y2="4"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</ul>
`;
